apps/avatax/src/modules/saleor/order-line.ts

```diff
@@ -0,0 +1,70 @@
+import { LineItemModel } from "avatax/lib/models/LineItemModel";
+import { SHIPPING_ITEM_CODE } from "../avatax/calculate-taxes/avatax-shipping-line";
+
+type SKU = string | null | undefined;
+type VariantId = string | null | undefined;
+
+export class SaleorOrderLine {
+  constructor(public taxIncluded: boolean) {
+    this.taxIncluded = taxIncluded;
+  }
```

Comment:
I think we should avoid passing data to domain layer both via constructor and methods.
I already want to refactor this.

Here is reasoning - we should use interfaces. Interface contains methods, not constructors. Constructor is a instance detail. 

If I want to inject "mock" OrderLine, I should be able to replace it with something following the same interface

Constructors on the other hands work great for dependency injection. So the idea is that we can create all instances we want this way:

```
new Order({
lines: [new OrderLine()]
})
```

and then, Order can technically use OrderLine to perform operation. But "taxIncluded" can be a detail of Order. If we need "taxIncluded" in constructor, we can't use dependency injection.

So I think you should move this param to `toAvataxLineItem` 

Alternative, but more complex way is 

```
class SaleorOrderLineGross extends SaleorOrderLine {}
class SaleorOrderLineNet extends SaleorOrderLine {}
```

but IMO this is premature and overcomplicated

Reply:
Alternative approach is using value object. In such case you can do this

```
// this is representation that contains data. no mapping
class OrderLine {
... fields

constructor(...fields) {}
}
```

```
class OrderLineConverter {
  saleorLineToAvataxLine(l: SaleorLine): LineItemModel
}
```

I think we should chose one of approaches.
Current one is a little "lying" because you create something that suggests is a Domain code, but it doesn't store any domain inside - its just a mapper.

So I suggest
1. Either use ValueObject + converter, 
2. or ValueObject + converting methods
3. Or just a mapper (then call it differently)

Reply:
I went into adding param to `toAvataxLineItem`. After refactor `SaleorOrderLine` should have it's own schema and be instantiated inside `SaleorOrder` so other places will use it 😄